### PR TITLE
[CMS-687] Make update-from-deprecated-upstream upgrade to composer-managed upstream.

### PR DESCRIPTION
**Pull request by kporras07**

### CODE REVIEW

File: src/Commands/ConvertToDrupalRecommendedSiteCommand.php [hunk 1]
```diff
@@ -187,6 +195,14 @@ private function updateComposerJsonMeta(string $localPath): void
         $composerJson['extra']['installer-paths']['web/profiles/custom/{$name}'] = ['type:drupal-custom-profile'];
         $composerJson['extra']['installer-paths']['web/themes/custom/{$name}'] = ['type:drupal-custom-theme'];
 
+        foreach (['autoload', 'scripts', 'scripts-description'] as $item) {
+            $composerJson[$item] = $composerManagedComposerJson[$item];
```
Comment (jabxjab) [inline src/Commands/ConvertToDrupalRecommendedSiteCommand.php:199]: Is there a possibility for `$composerManagedComposerJson` not to have some keys?
In other words, should this be like: 
```
if (isset($composerManagedComposerJson[$item])) {
  $composerJson[$item] = $composerManagedComposerJson[$item];
}
```

Comment (kporras07) [reply src/Commands/ConvertToDrupalRecommendedSiteCommand.php:199]: I don't think so, it's here: https://github.com/greg-1-anderson/drupal-composer-managed/blob/default/composer.json

File: src/Commands/ConvertToDrupalRecommendedSiteCommand.php [hunk 2]
```diff
@@ -119,12 +126,12 @@ public function convert(
...
         $this->getGit()->commit(
-            'Update composer.json to match "drupal-recommended" upstream and install dependencies',
+            'Update composer.json to match "drupal-composer-managed" upstream and install dependencies',
             ['composer.json', 'composer.lock']
         );
-        $this->log()->notice('composer.json updated to match "drupal-recommended" upstream');
+        $this->log()->notice('composer.json updated to match "drupal-composer-managed" upstream');
 
-        $this->detectDrupalProjectDiff($localPath);
+        $upstream_id === 'drupal9' ? $this->detectDrupalProjectDiff($localPath) : $this->detectDrupalRecommendedDiff($localPath);
```
Comment (greg-1-anderson) [inline src/Commands/ConvertToDrupalRecommendedSiteCommand.php:134]: Let's use an actual if / else here. Aliasing `else` to `:` isn't the same as eliminating an `else`. 😄

File: src/Commands/ConvertToDrupalRecommendedSiteCommand.php (hunk 1 above)
Comment (greg-1-anderson) [reply src/Commands/ConvertToDrupalRecommendedSiteCommand.php:199]: Seems safe enough like this

File: src/Commands/ConvertToDrupalRecommendedSiteCommand.php [hunk 3]
```diff
@@ -302,4 +318,77 @@ private function detectDrupalProjectDiff(string $localPath): void
...
+
+    /**
+     * Detects the differences between the site's code and "drupal-recommended" upstream code.
+     *
+     * If the differences are found, try to apply the patch and ask the user to resolve merge conflicts if found.
+     * Otherwise - apply the patch and commit the changes.
+     *
+     * @param string $localPath
+     */
+    private function detectDrupalRecommendedDiff(string $localPath): void
```
Comment (greg-1-anderson) [inline src/Commands/ConvertToDrupalRecommendedSiteCommand.php:330]: This function looks exactly the same as `detectDrupalProjectDiff, though, save for a couple of different constants. Maybe these two entrypoints could both call a common implementation that took these constants as parameters?

File: src/Commands/ConvertToDrupalRecommendedSiteCommand.php (hunk 1 above)
Comment (greg-1-anderson) [reply src/Commands/ConvertToDrupalRecommendedSiteCommand.php:199]: If we did add any special checking, I would put it in advance and fail fast, e.g. "drupal-composer-managed upstream missing expected portions of composer.json; can not continue, aborting." That might be prudent, and wouldn't require extra nesting in the loop.

**Review by greg-1-anderson — APPROVED** [review]
LGTM, suggested some optional refinements.

**Comment by kporras07** [comment]
Applied feedback, tests are passing. Merging

**Pull request merged**